pandas/types/cast.py

```diff
@@ -866,8 +866,23 @@ def _possibly_cast_to_datetime(value, dtype, errors='raise'):
 
 def _find_common_type(types):
     """Find a common data type among the given dtypes."""
-    # TODO: enable using pandas-specific types
+
+    if len(types) == 0:
+        raise ValueError('no types given')
+
+    first = types[0]
+    # workaround for find_common_type([np.dtype('datetime64[ns]')] * 2)
+    # => object
```

Comment:
this function can *almost* get rid of: https://github.com/pydata/pandas/blob/master/pandas/core/internals.py#L4799

though we use a bit more 'generic' types there